db/migrations/20210824111234_author_metrics.sql

```diff
@@ -0,0 +1,70 @@
+-- raw:
+--   environment
+--   event_type
+--   timestamp
+--   event
+
+-- timerEvents:
+--   path
+--   tabId
+--   userId
+--   seconds
+--   clientId
+--   increment
+
+-- #############################################################################
+-- migrate:up
+-- #############################################################################
+
+CREATE INDEX raw_timestamp ON raw (timestamp);
+CREATE INDEX raw_event_type_timestamp ON raw (event_type, timestamp);
+
+CREATE or replace FUNCTION get_post_id_from_path(path TEXT)
+RETURNS TEXT AS $$
+DECLARE
+  post_id TEXT;
+BEGIN
+  -- /posts/4fPxQjq6GFZgurSsf/room-for-other-things-how-to-adjust-if-ea-seems-overwhelming
+  -- /s/asdf/p/qwer
+  post_id := (SELECT (regexp_matches(path, '^/posts/([a-zA-Z0-9]+)/'))[1]);
+  if (post_id IS NOT NULL) THEN
+    RETURN post_id;
+  END IF;
+  post_id := (SELECT (regexp_matches(path, '^/s/[a-zA-Z0-9]+/p/([a-zA-Z0-9]+)'))[1]);
+  RETURN post_id;
+END
+$$ LANGUAGE plpgsql IMMUTABLE STRICT;
+
+CREATE INDEX raw_event_type_post_id ON raw (event_type, get_post_id_from_path(event->>'path'));
+
+-- Timer event is the main way to tell how many people have read your post and
```

Comment:
It's kind of cool to see code that I very definitively wrote being used and built upon. I hope you tested the timer event carefully though. I never validated the data.

Reply:
Good to know.